.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,59 @@
+name: SDK Benchmark Tests
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  sdk-benchmarks:
+    env:
+      py37: 3.7
+      py38: 3.8
+      py39: 3.9
+      py310: "3.10"
+      py311: "3.11"
+      pypy3: pypy-3.7
+      RUN_MATRIX_COMBINATION: ${{ matrix.python-version }}-sdk-${{ matrix.os }}
+    runs-on: self-hosted
+    strategy:
+      # Ensures the entire test matrix is run, even if one permutation fails
+      fail-fast: false
+      matrix:
+        python-version: [py37, py38, py39, py310, py311, pypy3]
+        os: [ubuntu-20.04, windows-2019]
+    steps:
+    - name: Checkout Core Repo @ SHA - ${{ github.sha }}
+      uses: actions/checkout@v2
+    - name: Set up Python ${{ env[matrix.python-version] }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env[matrix.python-version] }}
+        architecture: 'x64'
+    - name: Install tox
+      run: pip install tox==3.27.1 -U tox-factor
+    - name: Cache tox environment
+      # Preserves .tox directory between runs for faster installs
+      uses: actions/cache@v2
+      with:
+        path: |
+          .tox
+          ~/.cache/pip
+        key: v3-tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini',
+          'dev-requirements.txt') }}-core
+    - name: Run tox
+      run: tox -f ${{ matrix.python-version }}-sdk -- -k opentelemetry-sdk/tests/performance/benchmarks --benchmark-json=output.json
```

Comment:
Would it be possible to reduce the number of tests, or perhaps group them if there are tests that do the same thing with different parameters. I think that would make the charts much more usable. Java has many tests, but at this time only publishes results for a few.
https://github.com/open-telemetry/opentelemetry-java/blob/main/.github/workflows/benchmark.yml#L33
https://open-telemetry.github.io/opentelemetry-java/

Reply:
Could be but it seems a bit complex to modify the charts to produce one single chart with several data lines. We can address this issue later to avoid blocking all benchmarks in this PR.